elastic/kibana
# [RAM] Preserve relative snooze when adding or removing snooze schedules

## Summary

Closes #157868 

When adding or removing a snooze schedule, existing relative snoozes (including indefinite snoozes) will no longer be removed.

This was already how the bulk scheduling/unscheduling APIs behaved, so this PR brings the single snooze/unsnooze API into sync with the bulk.

### Testing

- Create a rule
- Add a relative snooze to this rule using the notification badge, e.g. 3 hours
- Add a snooze schedule for sometime in the future to this same rule
- Ensure the snooze schedule gets added but the relative snooze is still active
- Create another rule, and snooze it indefinitely
- Add a snooze schedule to this new rule
- Ensure the snooze schedule gets added but the indefinite snooze is still active
- Remove the snooze schedules from both these rules
- Ensure the snooze schedules get removed but the relative/indefinite snoozes are still active
